# log type
int16 type

# log string
string logMessage